### DISCUSSION TITLE
Z-Wave JS Ui does not re-establish connection if TCP drops

### DISCUSSION THREAD

**Discussion by goldriver2** [zwave-js/zwave-js#7201] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

I'm using a Z-Wave adapter (with Docker and Z-Wave-JS installed) shared via TCP (ser2net on the server side). my router rebooted today and the connection broke. However, it is not restored automatically as I had expected. Instead, the following message is displayed on the Z-Wave JS UI interface:
Driver: Serial port errored: read ETIMEDOUT (ZW0100) 

What did you expect to happen instead?
It would be nice if it tried to reconnect every few seconds until the server can be reached again. 

  1. Connect the Z-Net via ser2net
  2. simulate a network issue (disconnect router)
  3. reconnect router
  4. start ser2net

the message Driver: Serial port errored: read ETIMEDOUT (ZW0100) will be displayed and the z-Net is unreacheable

### Software versions

Driver (node-zwave-js): 13.1.0

Z-Wave JS UI: 9.17.0

Homeseer Z-Wave Plus plugin

### Z-Wave Controller (Stick/Dongle/...)

Manufacturer: Homeseer
Model name: N-net

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

2024-09-28 07:36:46.486 INFO Z-WAVE: Controller status: Driver: Serial port errored: read ECONNRESET (ZW0100)

2024-09-28 07:36:46.486 INFO GATEWAY: Driver is CLOSED

2024-09-28 07:36:46.487 INFO Z-WAVE: Restarting client in 1 seconds, retry 1

2024-09-28 07:36:47.503 INFO Z-WAVE-SERVER: Client disconnected

2024-09-28 07:36:47.515 INFO Z-WAVE-SERVER: Client disconnected

[a comment by an automated account is not shown]

**Comment by mangler99** [comment]
Same Issue - HomeSeer Z-Net + Home Assistant Workaround 

I'm experiencing the same issue with a **HomeSeer Z-Net (firmware v1.0.25)** connected via TCP to the Z-Wave JS UI addon in Home Assistant.                                                                                                                                                                       
                                                                                                                                                                                                                                                                                                                    
  ### My Setup                                                                                                                                                                                                                                                                                                      
  - Home Assistant OS                                                                                                                                                                                                                                                                                               
  - Z-Wave JS UI addon v6.1.2 (a0d7b954_zwavejs2mqtt)                                                                                                                                                                                                                                                               
  - HomeSeer Z-Net controller at `tcp://192.168.0.170:2001`                                                                                                                                                                                                                                                         
  - Static IP configured on the Z-Net                                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                    
  ### Symptoms                                                                                                                                                                                                                                                                                                      
  - Approximately once per month, the controller becomes unresponsive (still not sure the cause/trigger)                                                                                                                                                                                                                                               
  - Z-Wave JS shows "Controller is unresponsive"                                                                                                                                                                                                                                                                    
  - The Z-Net remains pingable and its web interface (port 80) is accessible                                                                                                                                                                                                                                        
  - The TCP serial port (2001) also remains open                                                                                                                                                                                                                                                                    
  - Restarting the Z-Wave JS addon resolves the issue immediately (no change or restart to the Z-Net)                                                                                                                                                                                                                                                   
  - The addon's Watchdog doesn't help because the addon itself is still "running" - it just lost the connection to the controller                                                                                                                                                                                   
                                                                                                                                                                                                                                                                                                                    
  ### Workaround - Home Assistant Automation                                                                                                                                                                                                                                                                        
Since the driver doesn't auto-reconnect, I created an automation that monitors the controller's node status sensor and automatically restarts the addon when it becomes unavailable:                                                                                                                              
                                                                                                                                                                                                                                                                                                                    
  ```yaml                                                                                                                                                                                                                                                                                                           
  - id: 'zwave_auto_recovery'                                                                                                                                                                                                                                                                                       
    alias: Z-Wave Controller Auto-Recovery                                                                                                                                                                                                                                                                          
    description: Automatically restarts Z-Wave JS addon when controller becomes unresponsive                                                                                                                                                                                                                        
    triggers:                                                                                                                                                                                                                                                                                                       
    - trigger: state                                                                                                                                                                                                                                                                                                
      entity_id: sensor.YOUR_CONTROLLER_node_status  # or sensor.YOUR_CONTROLLER_status                                                                                                                                                                                                                             
      to: unavailable                                                                                                                                                                                                                                                                                               
      for:                                                                                                                                                                                                                                                                                                          
        seconds: 60                                                                                                                                                                                                                                                                                                 
    conditions: []                                                                                                                                                                                                                                                                                                  
    actions:                                                                                                                                                                                                                                                                                                        
    - action: hassio.addon_restart                                                                                                                                                                                                                                                                                  
      data:                                                                                                                                                                                                                                                                                                         
        addon: a0d7b954_zwavejs2mqtt  # adjust to your addon slug                                                                                                                                                                                                                                                   
    - delay:                                                                                                                                                                                                                                                                                                        
        seconds: 30                                                                                                                                                                                                                                                                                                 
    - action: notify.mobile_app_YOUR_PHONE                                                                                                                                                                                                                                                                          
      data:                                                                                                                                                                                                                                                                                                         
        title: Z-Wave Auto-Recovery                                                                                                                                                                                                                                                                                 
        message: Z-Wave controller was unresponsive. Addon has been automatically restarted.                                                                                                                                                                                                                        
    mode: single                                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                    
  Replace sensor.YOUR_CONTROLLER_node_status with your controller's status entity (check in Developer Tools → States).                                                                                                                                                                                              
    For me this was sensor.ezzee_status   (Z-Net Controller Status)                    
                                                                                                                                                                                                                                                                                               
  This isn't a fix for the underlying issue, but it provides automatic recovery until proper TCP reconnection logic is implemented in the driver. Unfortunately, it also triggers a bunch of other automations ... in particular my water (leak) sensors all become "dry" again when the controller reconnects after restart. This causes my house to tell me about 15 water sensors have all cleared.  LOL!  It's funny to hear - but annoying at the same time.                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                    
  Request                                                                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                    
  Is there any timeline for implementing automatic reconnection attempts for TCP-based serial connections? The current behavior of "giving up" after a connection drop makes network-based controllers like the Z-Net unreliable without external workarounds.
